internal/datastore/postgres/postgres.go

```diff
@@ -122,7 +126,24 @@ func NewPostgresDatastore(
 	url string,
 	options ...Option,
 ) (datastore.Datastore, error) {
-	ds, err := newPostgresDatastore(ctx, url, options...)
+	ds, err := newPostgresDatastore(ctx, url, primaryInstanceID, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	return datastoreinternal.NewSeparatingContextDatastoreProxy(ds), nil
+}
+
+// NewReadOnlyPostgresDatastore initializes a SpiceDB datastore that uses a PostgreSQL
+// database by leveraging manual book-keeping to implement revisioning. This version is
+// read only and does not allow for write transactions.
+func NewReadOnlyPostgresDatastore(
+	ctx context.Context,
+	url string,
+	index uint32,
+	options ...Option,
+) (datastore.StrictReadDatastore, error) {
```

Comment:
I don't understand why what read-only has to do with being a strict datastore, those seem 2 orthogonal concerns. Why was this added?

Reply:
Strict depends on being read only and so I had it return both, since it fulfills both